### DISCUSSION TITLE
Disable preview

### DISCUSSION THREAD

**Discussion by StanBarrows** [laravel/nova-issues#3983] [Ideas & Feature Requests] [answered]
Is there any chance to disable the preview? Like with an additional policy?

<img width="1799" alt="CleanShot 2022-04-13 at 23 23 11@2x" src="https://user-images.githubusercontent.com/10268813/163272850-bbc7dd7a-cd60-4ea2-b173-69eb67034761.png">

**Comment by mdeimert** [comment]
Would be great to have this option.

**Comment by clementmas** [comment]
I think it should be disabled automatically if there are no fields to preview.

All my models show: "There are no fields to display". I'd even consider this a bug.

**Reply by vv-ac** [reply]
Same, I was also looking for that exact option. In more than 50% of the cases my resources do not need the preview option.

**Comment by orlyapps** [comment]
Some more ideas

- Trigger the preview with direct button and not hidden in the '...' dot menu. OR with a mouse hover?
- Within the preview modal "previous" <> "next"  resource so you can navigation through a lot of data
- Bonus: the preview modal as a slide over. So you can see list and the preview 
- In the preview modal a edit/show button

**Comment by FabrizioInfante** [comment]
I like the idea to add this with additionalPolicy like:

`    public function preview(Resource $resource)
`

**Comment by jaxramus** [comment]
is there actually no way to disable all previews entirely?

**Comment by ArtDepartmentMJ** [comment]
Also looking to disable the preview for a resource or globally

**Comment by nrueckmann** [comment]
My current solution is to just hide the button via css. It's not a real solution and i think all actions should be manageable by the consuming application, but it's still better than explaining each client why there is button without any function.

```
[dusk$="-preview-button"] {
  display: none;
}
```

**Reply by jaxramus** [reply]
this works great for now! thank you.

for anyone curious how to implement this, create a new CSS file in `resources/css/custom.css` with this content:

```css
[dusk$="-preview-button"] {
    display: none;
}
```

Then in your `app/Providers/NovaServiceProvider`, add this to the `boot` method:

```php
Nova::style('custom', resource_path('css/custom.css'));
```

**Reply by StanBarrows** [reply]
For me it was

```
[dusk$="-control-selector"] {
    display: none;
}
```

**Comment by fabiov** [comment]
Style css it's a work around, It would be better to have policy management-

**Reply by FabrizioInfante** [reply]
You are right. Also because if I hide preview button I have three dots that open nothing and it is very ugly

**Comment by RealityGamin** [comment]
This is marked as implemented but, I don't see a tag for what was implemented. Is there a way to disable it globally? Or make it so if there is nothing to preview, the button won't show?

**Reply by jaxramus** [reply]
> This is marked as implemented but, I don't see a tag for what was implemented. Is there a way to disable it globally? Or make it so if there is nothing to preview, the button won't show?

I noticed now if there are no fields set to preview, the menu option is hidden which is nice.

**Reply by RealityGamin** [reply]
You are right, awesome thank you!

This should be marked as completed.

**Comment by a-chaudhari** [comment]
I see this was marked as implemented and the prior comments say the item should auto-hide if no fields were marked as preview.  But that doesn't appear to be my case.

I still see the menu option for preview even if there are no fields marked as preview.  I'm on nova 4.12.13 which is the most current as of this post. 

Was there a regression? Or am I missing something?  Looks like in 4.12.2 this was added back according to the release notes.

**Reply by clementmas** [reply]
Did you update your assets? `php artisan nova:publish`. It works for me on the latest version.

**Reply by a-chaudhari** [reply]
doh.  that was it. thank you!

**Comment by davidhemphill** [comment, marked as answer]
This was implemented in Nova v4.9.0. 👍
